tests/integration/test_fim/test_files/test_whodata_policy_change/test_whodata_policy_change.py

```diff
@@ -0,0 +1,174 @@
+        - metadata:
+            type: dict
+            brief: Test case data.
+        - create_monitored_folders_module:
+            type: fixture
+            brief: Create the folders that will be monitored, delete them after test.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set ossec.conf configuration.
+        - configure_local_internal_options_function:
+            type: fixture
+            brief: Set local_internal_options configuration.
+        - restart_syscheck_function:
+            type: fixture
+            brief: restart syscheckd daemon, and truncate the ossec.log.
+        - wait_for_fim_start_function:
+            type: fixture
+            brief: check that the starting fim scan is detected.
+
+    assertions:
+        - Verify the SACL for the monitored files is configured
+        - Verify Whodata monitoring has started
+        - Verify that the event 4719 event is detected and changes monitoring to real-time
+        - Verify the monitoring mode changes to real-time
+        - Verify monitoring in real-time works correctly for the monitored files.
+
+    input_description:
+        - The file 'cases_whodata_policy_change.yaml' provides the test cases and specific configuration.
+        - The file 'configuration_whodata_policy_change.yaml' provides the configuration template to be used.
+
+    expected_output:
+        - fr".*win_whodata.*The SACL of '({file})' will be configured"
+        - r'.*win_whodata.*(Event 4719).*Switching directories to realtime'
+        - fr".*set_whodata_mode_changes.*The '({file})' directory starts to be monitored in real-time mode."
+        - r'.*Sending FIM event: (.+)$' ('added', 'modified', and 'deleted' events)
+
+    tags:
+        - whodata
+    '''
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+
+    # Check it is being monitored in whodata
+    evm.detect_windows_sacl_configured(wazuh_log_monitor, '.*')
+    # Check Whodata engine has started
+    evm.detect_whodata_start(wazuh_log_monitor)
+
+    # Change policies
+    if metadata['check_event']:
+        # Wait to allow thread_checker to be executed twice so Event 4719 detection starts.
+        time.sleep(6)
+    command = f"auditpol /restore /file:{os.path.join(TEST_DATA_PATH,metadata['disabling_file'])}"
+    output = run_local_command_returning_output(command)
+
+    # Check it changes to realtime
+    if metadata['check_event']:
+        evm.check_fim_event(timeout=20, callback=fim.CB_RECIEVED_EVENT_4719)
+    evm.detect_windows_whodata_mode_change(wazuh_log_monitor, '.*')
+
+    # Create/Update/Delete file and check events
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+    regular_file_cud(folder, wazuh_log_monitor, file_list=file_list, time_travel=False,
+                     event_mode=fim.REALTIME_MODE, min_timeout=global_parameters.default_timeout*4,
+                     triggers_event=True)
+
+    # Restore policies
```

Comment:
Add this to a fixture. The fixture will need to make a backup for the initial policies, and after finishing the test, it will have to restore those policies.